File: app/src/main/java/com/kazakago/cleanarchitecture/presentation/presenter/activity/AboutActivityViewModel.java
package com.kazakago.cleanarchitecture.presentation.presenter.activity;

import android.content.Context;

import com.kazakago.cleanarchitecture.presentation.listener.presenter.activity.AboutActivityViewModelListener;

/**
 * About Activity ViewModel
 *
 * @author Kensuke
 */
public class AboutActivityViewModel {

    private Context context;
    private AboutActivityViewModelListener listener;

    public AboutActivityViewModel(Context context, AboutActivityViewModelListener listener) {
        this.context = context;
        this.listener = listener;
    }

    public void onClickBackIcon() {
        listener.onPerformFinish();
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/presentation/presenter/activity/AboutActivityViewModel.kt
package com.kazakago.cleanarchitecture.presentation.presenter.activity

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LifecycleObserver
import com.kazakago.cleanarchitecture.presentation.listener.activity.AboutActivityViewModelListener

class AboutActivityViewModel(application: Application) : AndroidViewModel(application), LifecycleObserver {

    var listener: AboutActivityViewModelListener? = null

    fun onClickBackIcon() {
        listener?.finish()
    }

}
